api/python/quilt3/data_transfer.py

```diff
@@ -856,6 +861,76 @@ def calculate_sha256(src_list: List[PhysicalKey], sizes: List[int]):
     return _calculate_sha256_internal(src_list, sizes, [None] * len(src_list))
 
 
+async def _calculate_hash_s3_get_parts(s3_client_provider, s3_executor, s3_pending_parts_semaphore, src, size):
```

Comment:
consider `, *,` to make the last ~3 args mandatory kwargs, thus avoiding bad positional calls